llvm/test/CodeGen/AArch64/sve-fixed-length-partial-reduce.ll

```diff
@@ -407,6 +407,33 @@ define <4 x i32> @four_way_i8_i32_vl128(ptr %accptr, ptr %uptr, ptr %sptr) {
   ret <4 x i32> %partial.reduce
 }
 
+define <4 x i32> @four_way_i8_i32_vl128_usdot(ptr %accptr, ptr %uptr, ptr %sptr) {
```

Comment:
Could you also add a test where the zext and sext are reversed? (to ensure that we still get the usdot)
And perhaps a negative test where the element types are such that they don't result in a usdot (e.g. i8 -> i64).